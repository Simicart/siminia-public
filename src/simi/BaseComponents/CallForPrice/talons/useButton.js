import { useMemo, useCallback } from 'react';
import { getHidePriceEnable } from '../utils';
import { useAppContext } from '@magento/peregrine/lib/context/app';

export const useButton = (props = {}) => {
    const { product } = props;

    const { advancedhideprice, sku } = product || {};
    const [, { toggleDrawer }] = useAppContext();
    const hidePriceEnable = getHidePriceEnable();

    const isHidePrice = useMemo(() => {
        return (
            hidePriceEnable &&
            advancedhideprice &&
            advancedhideprice.advancedhideprice_type === 'hideprice'
        );
    }, [hidePriceEnable, advancedhideprice]);

    const isCallForPrice = useMemo(() => {
        return (
            hidePriceEnable &&
            advancedhideprice &&
            advancedhideprice.advancedhideprice_type === 'callforprice'
        );
    }, [hidePriceEnable, advancedhideprice]);

    const text = useMemo(() => {
        return advancedhideprice?.advancedhideprice_text;
    }, [advancedhideprice]);

    const handleOpen = useCallback(() => {
        toggleDrawer('hidePrice.form.' + sku);
    }, [toggleDrawer]);

    return {
        isHidePrice,
        isCallForPrice,
        text,
        productSku: sku,
        handleOpen
    };
};
