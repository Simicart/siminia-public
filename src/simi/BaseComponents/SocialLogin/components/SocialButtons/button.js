import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { shape, string, func } from 'prop-types';

import defaultClasses from './socialButtons.module.css';
import SocialIcon from '../SocialIcon';
import { FormattedMessage } from 'react-intl';

const Button = props => {
  const { label, type, url, handleLogin, buttonShape } = props;
  const classes = mergeClasses(defaultClasses, props.classes);
  const isRectangleShape = buttonShape === 'rectangle';

  const btnText = isRectangleShape ? (
    <span className={classes.text}>
      <FormattedMessage
        id={'amSocialLogin.continueWith'}
        defaultMessage={'Continue with {label}'}
        values={{ label }}
      />
    </span>
  ) : null;

  const style = {
    '--amslBtnBg': `var(--amsl-${type}-button__${
      isRectangleShape ? 'rectangle__' : ''
    }background)`
  };

  return (
    <button
      style={style}
      type="button"
      title={`Sign in with ${label}`}
      className={`${classes.button} ${classes[type]}`}
      onClick={() => handleLogin({ type, url })}
    >
      <SocialIcon type={type} />
      {btnText}
    </button>
  );
};

Button.propTypes = {
  classes: shape({
    root: string
  }),
  label: string,
  type: string,
  url: string,
  buttonShape: string,
  handleLogin: func
};

export default Button;
