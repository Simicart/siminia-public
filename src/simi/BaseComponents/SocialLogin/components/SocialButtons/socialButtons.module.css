:root {
  --amsl-google-button__background: #fff;
  --amsl-google-button__rectangle__background: #448fff;
  --amsl-facebook-button__background: #3c5a99;
  --amsl-facebook-button__rectangle__background: #3c5a99;
  --amsl-twitter-button__background: #1da1f2;
  --amsl-twitter-button__rectangle__background: var(
    --amsl-twitter-button__background
  );
  --amsl-linkedin-button__background: #0077b7;
  --amsl-linkedin-button__rectangle__background: var(
    --amsl-linkedin-button__background
  );
  --amsl-instagram-button__background: linear-gradient(
    0deg,
    #e09b3d,
    #c74c4d,
    #c21975,
    #7024c4
  );
  --amsl-instagram-button__rectangle__background: var(
    --amsl-instagram-button__background
  );
  --amsl-amazon-button__background: #fff;
  --amsl-amazon-button__rectangle__background: #232f3e;
  --amsl-paypal-button__background: #fff;
  --amsl-paypal-button__rectangle__background: #009cde;
  --amsl-twitch-button__background: #6441a5;
  --amsl-twitch-button__rectangle__background: var(
    --amsl-twitch-button__background
  );
  --amsl-apple-button__background: #fff;
  --amsl-apple-button__rectangle__background: var(
    --amsl-apple-button__background
  );
}

.root {
  padding: 2rem;
  background: rgb(var(--venia-global-color-gray));
}

.list {
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  justify-content: center;
}

.account .list {
  justify-content: flex-start;
}

.button {
  --amslBtnBg: --amsl-google-button__background;
  display: block;
  margin: 0.5rem;
  width: 4rem;
  height: 4rem;
  background: var(--amslBtnBg);
  border-radius: 0.5rem;
  transition: opacity 0.3s;
  box-shadow: 0 3px 6px rgb(var(--venia-global-color-gray-900) / 0.2);
  overflow: hidden;
}

.button:hover {
  box-shadow: 0 3px 6px rgb(var(--venia-global-color-gray-900) / 0.1);
  opacity: 0.9;
}

.round .button {
  border-radius: 4rem;
}

.rectangle {
  display: block;
}

.rectangle .button {
  width: 100%;
  height: auto;
  display: flex;
  align-items: center;
  color: #fff;
  margin: 0;
  padding: 0.5rem 1rem 0.5rem 0.5rem;
  border-radius: 0.25rem;
  min-width: 0;
}

.rectangle .button + .button {
  margin-top: 1rem;
}

.button.apple {
  color: rgb(var(--venia-global-color-text));
}

.text {
  margin-left: 1rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: block;
}

.showMore {
  display: block;
  margin: 1.25rem auto 0 auto;
  color: rgb(var(--venia-brand-color-1-700));
  font-size: var(--venia-global-fontSize-200);
  line-height: 1.25rem;
  cursor: pointer;
}

.showMore:hover {
  text-decoration: underline;
}

.showMore:after {
  content: '';
  display: inline-block;
  width: 0.5rem;
  height: 0.5rem;
  border-left: 1px solid rgb(var(--venia-brand-color-1-700));
  border-bottom: 1px solid rgb(var(--venia-brand-color-1-700));
  transform: rotate(-45deg);
  margin-left: 0.5rem;
  position: relative;
  top: -2px;
}

.message {
  display: block;
  margin-top: 0.5rem;
  color: rgb(var(--venia-global-color-error));
  font-size: 0.875rem;
  font-weight: var(--venia-global-fontWeight-semibold);
  line-height: 1rem;
  padding: 0.625rem 0.125rem 0.125rem;
}
