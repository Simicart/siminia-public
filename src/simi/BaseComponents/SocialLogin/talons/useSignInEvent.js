import SIGNIN_OPERATIONS from '@magento/peregrine/lib/talons/SignIn/signIn.gql.js';
import { GET_CART_DETAILS_QUERY as getCartDetailsQuery } from '@magento/venia-ui/lib/components/SignIn/signIn.gql.js';
import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
import { useApolloClient, useMutation } from '@apollo/client';
import { useCallback, useState } from 'react';
import { useCartContext } from '@magento/peregrine/lib/context/cart';
import { useUserContext } from '@magento/peregrine/lib/context/user';
import { useAwaitQuery } from '@magento/peregrine/lib/hooks/useAwaitQuery';
import { clearCartDataFromCache } from '@magento/peregrine/lib/Apollo/clearCartDataFromCache';
import { clearCustomerDataFromCache } from '@magento/peregrine/lib/Apollo/clearCustomerDataFromCache';
import { retrieveCartId } from '@magento/peregrine/lib/store/actions/cart';
import { useEventListener } from '@magento/peregrine';

export const useSignInEvent = (props = {}) => {
  const { handleErrors } = props;
  const operations = mergeOperations(SIGNIN_OPERATIONS, props.operations);

  const {
    createCartMutation,
    getCustomerQuery,
    mergeCartsMutation
  } = operations;

  const apolloClient = useApolloClient();
  const [{ isSignedIn }] = useUserContext();
  const [isSigningIn, setIsSigningIn] = useState(false);

  const [
    { cartId },
    { createCart, removeCart, getCartDetails }
  ] = useCartContext();

  const [{ isGettingDetails }, { getUserDetails, setToken }] = useUserContext();

  const [fetchCartId] = useMutation(createCartMutation);
  const [mergeCarts] = useMutation(mergeCartsMutation);
  const fetchUserDetails = useAwaitQuery(getCustomerQuery);
  const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);

  const socialLoginEventHandler = useCallback(
    async event => {
      setIsSigningIn(true);

      try {
        if (isSignedIn) {
          return null;
        }

        const { token, result, messages, redirect_data } = event.data || {};

        if (result === 1 && token) {
          const sourceCartId = cartId;

          await setToken(token);

          // Clear all cart/customer data from cache and redux.
          await clearCartDataFromCache(apolloClient);
          await clearCustomerDataFromCache(apolloClient);
          await removeCart();
          // Create and get the customer's cart id.
          await createCart({
            fetchCartId
          });
          const destinationCartId = await retrieveCartId();

          // Merge the guest cart into the customer cart.
          await mergeCarts({
            variables: {
              destinationCartId,
              sourceCartId
            }
          });

          // Ensure old stores are updated with any new data.
          getUserDetails({ fetchUserDetails });
          getCartDetails({ fetchCartId, fetchCartDetails });
        } else if (result === 0) {
          handleErrors({
            message: messages[0] || null,
            redirectWithError: redirect_data && redirect_data.redirectWithError
          });
        }

        setIsSigningIn(false);
      } catch (error) {
        if (process.env.NODE_ENV !== 'production') {
          console.error(error);
        }

        setIsSigningIn(false);
      }
    },
    [
      cartId,
      apolloClient,
      removeCart,
      setToken,
      createCart,
      fetchCartId,
      mergeCarts,
      getUserDetails,
      fetchUserDetails,
      getCartDetails,
      fetchCartDetails,
      isSignedIn,
      handleErrors
    ]
  );

  useEventListener(window, 'message', socialLoginEventHandler);

  return {
    isBusy: isGettingDetails || isSigningIn
  };
};
