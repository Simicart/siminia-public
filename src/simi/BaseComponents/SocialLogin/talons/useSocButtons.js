import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useSocialLoginContext } from '../context';
import { useUserContext } from '@magento/peregrine/lib/context/user';

const VISIBLE_BUTTONS_COUNT = 3;

const getPopupParams = () => {
  const bottomPadding = 22;

  const screenX = window.screenX || window.screenLeft;
  const screenY = window.screenY || window.screenTop;
  const outerWidth = window.outerWidth || document.body.clientWidth;
  const outerHeight =
    window.outerHeight || document.body.clientHeight - bottomPadding;

  const width = 500;
  const height = 420;

  const leftDivider = 2;
  const topDivider = 2.5;

  const left = parseInt(screenX + (outerWidth - width) / leftDivider, 10);
  const top = parseInt(screenY + (outerHeight - height) / topDivider, 10);

  return `width=${width},height=${height},left=${left},top=${top}`;
};

export const useSocButtons = (props = {}) => {
  const { buttons, mode, showCreateAccount } = props;
  const isShowMoreBtn =
    mode === 'popup' && buttons.length > VISIBLE_BUTTONS_COUNT;
  const [showAll, setShowAll] = useState(!isShowMoreBtn);
  const { buttonShape, errors, handleErrors } = useSocialLoginContext();
  const [{ isSignedIn, token }] = useUserContext();
  const errorRef = useRef(null);

  const visibleButtons = useMemo(
    () => (showAll ? buttons : buttons.slice(0, VISIBLE_BUTTONS_COUNT)),
    [showAll, buttons]
  );

  const handleLogin = useCallback(
    button => {
      const { type, url } = button;

      handleErrors(null);

      window.open(
        `${url}&isAjax=true${
          !isSignedIn ? '&generateToken=true' : `&token=${token}`
        }`,
        type,
        getPopupParams()
      );
    },
    [isSignedIn, handleErrors, token]
  );

  const handleShowMore = useCallback(() => setShowAll(true), [setShowAll]);

  useEffect(() => {
    const el = errorRef && errorRef.current;

    if (errors) {
      const { redirectWithError } = errors;

      if (redirectWithError && showCreateAccount) {
        handleErrors({
          ...errors,
          redirectWithError: 0
        });

        showCreateAccount();
      }

      if (el && mode === 'popup') {
        el.scrollIntoView();
      }
    }

    return () => {
      if (errors && !errors.redirectWithError) {
        handleErrors(null);
      }
    };
  }, [errors, errorRef, showCreateAccount, handleErrors, mode]);

  return {
    handleLogin,
    visibleButtons,
    isShowMoreBtn: isShowMoreBtn && !showAll,
    handleShowMore,
    buttonShape,
    errors,
    errorRef
  };
};
