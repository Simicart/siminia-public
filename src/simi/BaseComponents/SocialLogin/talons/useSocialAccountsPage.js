import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
import DEFAULT_OPERATIONS from './socialLogin.ggl';
import { useLazyQuery, useMutation } from '@apollo/client';
import { useCallback, useEffect, useMemo } from 'react';
import { useUserContext } from '@magento/peregrine/lib/context/user';
import { useHistory } from 'react-router-dom';
import { useSocialLoginContext } from '../context';
import { useEventListener } from '@magento/peregrine';

export const useSocialAccountsPage = (props = {}) => {
  const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
  const { getSocAccountDataQuery, unlinkAccountMutation } = operations;

  const {
    buttons: allButtons,
    handleErrors,
    isEnabled
  } = useSocialLoginContext();
  const [{ isSignedIn }] = useUserContext();
  const history = useHistory();

  const [runQuery, { data, loading }] = useLazyQuery(getSocAccountDataQuery, {
    fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first',
    skip: !isSignedIn || !isEnabled
  });

  const { amSocialLoginAccountData: linkedAccounts } = data || {};

  const [unlinkAccount, { loading: unlinkLoading }] = useMutation(
    unlinkAccountMutation
  );

  const handleUnlink = useCallback(
    async type => {
      try {
        await unlinkAccount({
          variables: { type },
          refetchQueries: [{ query: getSocAccountDataQuery }],
          awaitRefetchQueries: true
        });
      } catch (error) {
        if (process.env.NODE_ENV === 'development') {
          console.error(error);
        }
      }
    },
    [unlinkAccount, getSocAccountDataQuery]
  );

  // If the user is no longer signed in, redirect to the home page.
  useEffect(() => {
    if (!isSignedIn) {
      history.push('/');
    } else {
      runQuery();
    }
  }, [history, isSignedIn, runQuery]);

  const buttons = useMemo(() => {
    if (!allButtons || !linkedAccounts) {
      return null;
    }

    return allButtons.filter(btn =>
      linkedAccounts.every(({ type }) => type !== btn.type)
    );
  }, [linkedAccounts, allButtons]);

  const linkAccountHandler = useCallback(
    event => {
      try {
        if (!isSignedIn) {
          return null;
        }

        const { result, messages, token } = event.data || {};

        if (result === 1 && !token) {
          handleErrors(null);
          runQuery();
        } else if (messages) {
          handleErrors({
            message: messages[0]
          });
        }
      } catch (e) {
        if (process.env.NODE_ENV !== 'production') {
          console.error(error);
        }
      }
    },
    [runQuery, handleErrors, isSignedIn]
  );

  useEventListener(window, 'message', linkAccountHandler);

  return {
    isEnabled,
    linkedAccounts,
    loading,
    handleUnlink,
    buttons,
    isDisabled: unlinkLoading
  };
};
