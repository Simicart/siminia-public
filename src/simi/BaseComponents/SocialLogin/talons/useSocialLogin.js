import { useCallback, useEffect, useRef, useState } from 'react';
import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
import DEFAULT_OPERATIONS from './socialLogin.ggl';
import { useQuery } from '@apollo/client';
import { BUTTON_SHAPE } from '../constants';
import { useUserContext } from '@magento/peregrine/lib/context/user';
import { useHistory } from 'react-router-dom';

export const useSocialLogin = (props = {}) => {
  const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);

  const { getStoreConfigQuery, getSocButtonsConfigsQuery } = operations;
  const [{ isSignedIn }] = useUserContext();
  const prevIsSignedIn = useRef(isSignedIn);
  const history = useHistory();

  const { data: storeConfigData } = useQuery(getStoreConfigQuery, {
    fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first'
  });

  const { data: buttonsConfigData } = useQuery(getSocButtonsConfigsQuery, {
    fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first'
  });

  const [errors, setErrors] = useState(null);

  const handleErrors = useCallback(error => setErrors(error), [setErrors]);

  const { storeConfig } = storeConfigData || {};
  const { amSocialLoginButtonConfig: buttons } = buttonsConfigData || {};

  const {
    amsociallogin_general_enabled: isEnabled,
    amsociallogin_general_button_shape,
    amsociallogin_general_button_position,
    amsociallogin_general_redirect_type: redirectType,
    amsociallogin_general_custom_url: redirectUrl,
    amsociallogin_general_login_position,
    amsociallogin_general_popup_enabled: popupEnabled
  } = storeConfig || {};

  const enabledModes =
    amsociallogin_general_login_position &&
    amsociallogin_general_login_position.split(',');

  const needRedirect = redirectType === 1 && redirectUrl;

  useEffect(() => {
    if (
      popupEnabled &&
      isSignedIn &&
      prevIsSignedIn.current !== isSignedIn &&
      needRedirect
    ) {
      history.push(`/${redirectUrl.replace(/^\//, '')}`);
    }

    prevIsSignedIn.current = isSignedIn;
  }, [
    needRedirect,
    prevIsSignedIn,
    isSignedIn,
    redirectUrl,
    history,
    popupEnabled
  ]);

  return {
    storeConfig,
    buttons,
    isEnabled: isEnabled && Array.isArray(buttons) && buttons.length,
    buttonShape: BUTTON_SHAPE[amsociallogin_general_button_shape || 0],
    buttonPosition: amsociallogin_general_button_position,
    needRedirect,
    errors,
    handleErrors,
    enabledModes
  };
};
