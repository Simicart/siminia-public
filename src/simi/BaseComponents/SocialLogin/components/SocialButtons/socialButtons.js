import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { shape, string, array } from 'prop-types';

import defaultClasses from './socialButtons.module.css';
import Button from './button';
import { useSocButtons } from '../../talons/useSocButtons';
import { FormattedMessage } from 'react-intl';

const SocialButtons = props => {
  const { buttons, mode } = props;
  const {
    buttonShape,
    visibleButtons,
    handleLogin,
    isShowMoreBtn,
    handleShowMore,
    errors,
    errorRef
  } = useSocButtons({ ...props });

  if (!Array.isArray(buttons) || !buttons.length) {
    return null;
  }
  const classes = mergeClasses(defaultClasses, props.classes);

  const list = visibleButtons.map(btn => (
    <Button
      key={btn.type}
      buttonShape={buttonShape}
      {...btn}
      handleLogin={handleLogin}
    />
  ));

  const showMoreBtn = isShowMoreBtn ? (
    <button className={classes.showMore} onClick={handleShowMore}>
      <FormattedMessage
        id={'amSocialLogin.showMore'}
        defaultMessage={'Show More'}
      />
    </button>
  ) : null;

  const errorMessage =
    errors && errors.message ? (
      <span ref={errorRef} className={classes.message}>
        {errors.message}
      </span>
    ) : null;

  return (
    <div className={`${classes.root} ${classes[buttonShape]} ${classes[mode]}`}>
      <div className={classes.list}>{list}</div>
      {showMoreBtn}
      {errorMessage}
    </div>
  );
};

SocialButtons.propTypes = {
  buttons: array,
  classes: shape({
    root: string
  })
};

export default SocialButtons;
