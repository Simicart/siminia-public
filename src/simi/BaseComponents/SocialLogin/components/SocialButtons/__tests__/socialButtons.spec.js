import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import SocialButtons from '../socialButtons';
import { useSocButtons } from '../../../talons/useSocButtons';

jest.mock('../button', () => 'Button');
jest.mock('../../../talons/useSocButtons');
jest.mock('react-intl', () => ({
  FormattedMessage: props => (
    <div componentName="Formatted Message Component" {...props} />
  ),
  useIntl: jest.fn().mockReturnValue({
    formatMessage: jest
      .fn()
      .mockImplementation(options => options.defaultMessage)
  })
}));
jest.mock('../../../context', () => {
  const useSocialLoginContext = jest.fn(() => {});

  return { useSocialLoginContext };
});

const buttons = [
  {
    label: 'Facebook',
    type: 'facebook'
  },
  {
    label: 'Instagram',
    type: 'instagram'
  },
  {
    label: 'LinkedIn',
    type: 'linkedin'
  },
  {
    label: 'Google',
    type: 'google'
  }
];

test('render empty', () => {
  useSocButtons.mockReturnValue({ visibleButtons: buttons });

  const tree = createTestInstance(<SocialButtons buttons={null} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render without show more btn', () => {
  useSocButtons.mockReturnValue({
    visibleButtons: buttons.slice(0, 2),
    isShowMoreBtn: false
  });

  const tree = createTestInstance(
    <SocialButtons buttons={buttons.slice(0, 2)} />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render with show more btn', () => {
  useSocButtons.mockReturnValue({
    visibleButtons: buttons,
    isShowMoreBtn: true
  });

  const tree = createTestInstance(<SocialButtons buttons={buttons} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render with error', () => {
  useSocButtons.mockReturnValue({
    visibleButtons: buttons,
    isShowMoreBtn: true,
    errors: {
      message: 'Error message'
    }
  });

  const tree = createTestInstance(<SocialButtons buttons={buttons} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
