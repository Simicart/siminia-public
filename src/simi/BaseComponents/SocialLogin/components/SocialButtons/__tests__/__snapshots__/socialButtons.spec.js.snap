// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty 1`] = `null`;

exports[`render with error 1`] = `
<div
  className="undefined undefined undefined"
>
  <div>
    <Button
      label="Facebook"
      type="facebook"
    />
    <Button
      label="Instagram"
      type="instagram"
    />
    <Button
      label="LinkedIn"
      type="linkedin"
    />
    <Button
      label="Google"
      type="google"
    />
  </div>
  <button>
    <div
      componentName="Formatted Message Component"
      defaultMessage="Show More"
      id="amSocialLogin.showMore"
    />
  </button>
  <span>
    Error message
  </span>
</div>
`;

exports[`render with show more btn 1`] = `
<div
  className="undefined undefined undefined"
>
  <div>
    <Button
      label="Facebook"
      type="facebook"
    />
    <Button
      label="Instagram"
      type="instagram"
    />
    <Button
      label="LinkedIn"
      type="linkedin"
    />
    <Button
      label="Google"
      type="google"
    />
  </div>
  <button>
    <div
      componentName="Formatted Message Component"
      defaultMessage="Show More"
      id="amSocialLogin.showMore"
    />
  </button>
</div>
`;

exports[`render without show more btn 1`] = `
<div
  className="undefined undefined undefined"
>
  <div>
    <Button
      label="Facebook"
      type="facebook"
    />
    <Button
      label="Instagram"
      type="instagram"
    />
  </div>
</div>
`;
